extensions/merge-conflict/src/mergeConflictParser.ts

```diff
@@ -49,7 +51,11 @@ export class MergeConflictParser {
 				}
 
 				// Create a new conflict starting at this line
-				currentConflict = { startHeader: line };
+				currentConflict = { startHeader: line, commonAncestors: null };
+			}
+			// Are we within a conflict block and is this a common ancestors marker? |||||||
+			else if (currentConflict && line.text.startsWith(commonAncestorsMarker)) {
+				currentConflict.commonAncestors = line;
```

Comment:
If, for some reason, we ended up with:

```
<<<<<<< HEAD
Foo
||||||| common
foo
||||||| common
fooo
=======
blah
>>>>>>> incoming
```

We will end up extending the "current" content over the first merge ancestor block. If we encounter two blocks of `|||||||` I think there are two thing we could do: 

 - Favorable: Don't overwrite `currentConflict.commonAncestors`, just leave it with the first occurrence, this way the current content block will span the correct range. (e.g. `else if (currentConflict && !currentConflict.commonAncestors && line.text.startsWith(commonAncestorsMarker))`
 - Break out of scanning (like we do if we encounter a start marker before an end marker), so we don't provide any decoration if we are unsure what to do. 

FWIW I think this is also true of existing `=======` scanning...